prov/lnx/src/lnx_ops.c

```diff
@@ -699,35 +637,31 @@ ssize_t lnx_tsenddata(struct fid_ep *ep, const void *buf, size_t len, void *desc
 {
 	int rc;
 	struct lnx_ep *lep;
-	struct lnx_peer *lp;
-	struct local_prov_ep *cep;
+	struct lnx_core_ep *cep;
 	fi_addr_t core_addr;
-	struct lnx_peer_table *peer_tbl;
-	void *mem_desc;
-	struct ofi_mr_entry *mre = NULL;
-	struct iovec iov = {.iov_base = (void*)buf, .iov_len = len};
+	void *core_desc = desc;
 
-	lep = lnx_get_lep(ep, NULL);
+	lep = container_of(ep, struct lnx_ep, le_ep.ep_fid.fid);
 	if (!lep)
 		return -FI_ENOSYS;
 
-	peer_tbl = lep->le_peer_tbl;
-
-	lp = lnx_av_lookup_addr(peer_tbl, dest_addr);
-	rc = lnx_select_send_pathway(lp, lep->le_domain, desc, &cep,
-				&core_addr, &iov, 1, &mre, &mem_desc, NULL);
+	rc = lnx_select_send_endpoints(lep, dest_addr, &cep, &core_addr);
 	if (rc)
 		return rc;
 
 	FI_DBG(&lnx_prov, FI_LOG_CORE,
 	       "sending to %lx tag %lx buf %p len %ld\n",
 	       core_addr, tag, buf, len);
 
-	rc = fi_tsenddata(cep->lpe_ep, buf, len, mem_desc,
-			  data, core_addr, tag, context);
+	/* do memory registration with the core domain */
```

Comment:
I don't think any of these comments are necessary since the function is pretty self explanatory.